test/test_module_init.py

```diff
@@ -0,0 +1,330 @@
+import inspect
+import torch
+import unittest
+from unittest.mock import MagicMock, patch
+from torch.testing._internal.common_utils import TestCase, run_tests
+from torch.testing._internal.common_device_type import instantiate_device_type_tests
+
+
+# Returns a database of args & kwargs that can be used to construct each module.
+# Each entry is in class -> (args, kwargs) format.
+# Example: torch.nn.Linear -> ([10, 5], {})
+def build_constructor_arg_db():
+    return {
+        torch.nn.AdaptiveAvgPool1d: ([5], {}),
```

Comment:
If someone adds a new module, will they know all the lists they need to add it to?

Reply:
Any module in `__all__` for torch.nn modules (and the various quantization modules) will be tested here and an error will be thrown indicating that an entry should be added here.